src/server/_common.py

```diff
@@ -18,24 +21,63 @@ def _get_db() -> Connection:
         g.db = conn
     return g.db
 
+def log_and_raise_validation_exception(message):
```

Comment:
raising and not catching `werkzeug.exceptions.HTTPException` look like they actually produce http responses in themselves (see https://werkzeug.palletsprojects.com/en/2.2.x/exceptions/ ).  all the instances of these that we create in the server code are immediately raised, so the logger statement could go into the `_exceptions.EpiDataException` base class and remove the need for this method.

alternatively, in what might be a cleaner implementation, we can catch them all and log them in one place with this: https://flask.palletsprojects.com/en/2.2.x/errorhandling/#generic-exception-handlers